courses/models.py

```diff
@@ -343,3 +347,31 @@ class APIKey(models.Model):
     active = models.BooleanField(blank=True, default=True)
 
     privileges = models.ManyToManyField(APIPrivilege, related_name='key_set', blank=True)
+
+
+class UserData(models.Model):
+    """
+    A model that stores all user data from PCX users
+    """
+    user = models.OneToOneField(get_user_model(), on_delete=models.CASCADE, related_name='user')
+    email = models.EmailField(blank=True, null=True)
+
+    def validate_phone(value):
+        try:
+            phonenumbers.parse(value, 'US')
+        except phonenumbers.phonenumberutil.NumberParseException:
+            raise ValidationError('Enter a valid phone number.')
+
+    phone = models.CharField(blank=True, null=True, max_length=100, validators=[validate_phone])
```

Comment:
all fields should be together at the top of a model class for readability

Reply:
The problem is validate_phone has to be defined above in Python... also lambda functions don't allow imperative programming at all so I couldn't do it that way.  Is there a better workaround or can I just add a comment that the phone field is below?

Reply:
Good point! Keeping it as-is is good.